xfp/xlist.py

```diff
@@ -75,20 +77,20 @@ def head(self) -> X:
 
     def tail(self) -> "Xlist[X]":
         """Return the Xlist / its first element.
-        
+
         ### Raise
-        
+
         - IndexError -- if the list is empty.
         """
         if len(self) <= 0:
             raise IndexError("<tail> operation not allowed on empty list")
         return Xlist(self.__data[1:])
 
-    def map(self, f: Callable[[X], E]) -> "Xlist[E]":
+    def map(self, f: Callable[[X], Y]) -> "Xlist[Y]":
```

Comment:
on a un choix à faire : ou bien f: X => E partout, et E est la variable générique pour "chépa", ou bien si tu veux un générique par fonction mais dans ce cas, python 3.12 autorise à la place du typeVar
```suggestion
    def map[Y](self, f: Callable[[X], Y]) -> "Xlist[Y]":
```
Ce qui me gêne avec cette façon de faire c'est qu'on pourra jamais être homogène partout dans la mesure ou la feature générique n'est pas complète (entre autre high order types absents). Apres si en faisant la modif on se rend compte que ça pose pas de problème, pourquoi pas

Reply:
en essayant de faire machin[Y] un peu partout, avis un peu plus argumenté sur la question : l'existence de `def flatten(self) -> "Xlist[E]"` rend la chose hyper compliqué, je te propose de rester sur un type E inconnu pour le moment et d'aviser plus tard

tl;dr; si tu as pas le temps d'y réfléchir tout de suite, on reste sur E pendant mon interco pour me donner un jouet et on verra plus tard